src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -1076,7 +1076,15 @@ protected async Task<TResult> InvokeCoreAsync<TResult>(long? id, string targetNa
                                     this.TraceSource.TraceEvent(TraceEventType.Warning, (int)TraceEvents.RequestAbandonedByRemote, "Aborting pending request \"{0}\" because the connection was lost.", id);
                                 }
 
-                                tcs.TrySetException(new ConnectionLostException());
+                                if (cancellationToken.IsCancellationRequested || this.IsDisposed)
```

Comment:
The ` || this.IsDisposed` is beyond the scope of #265 as I understand it. And if we're careful about which exception we throw in that case, perhaps we should be throwing `ObjectDisposedException` instead since the object was disposed while methods were still running inside the local instance.

Reply:
I removed the disposal check.